go-rock/1.22-24.04/rockcraft.yaml

```
@@ -0,0 +1,54 @@
+name: go
+build-base: ubuntu@24.04
+base: bare
+version: '1.22'
+summary: The Go programming language
+description: |
+  Go is an open source programming language that enables the production of
+  simple, efficient and reliable software at scale.
+platforms: 
+  amd64:
+  arm64:
+
+# We are not using the _daemon_ run-user, since rockcraft adds this user
+# locked. We however needs sudo to run apt.
+# https://github.com/canonical/rockcraft/blob/acc25f85af3ec691eb7f9f41558164745c03e028/rockcraft/oci.py#L297
+
```

```suggestion

environment:
  GOTOOLCHAIN: local
  GOLANG_VERSION: 1.22.2

```

https://go.dev/doc/toolchain

to encourage go to actually use the go in this rock?

---

prior art:

```bash
docker run --rm golang:trixie go env GOTOOLCHAIN
```

aka the official golang image prints `local`

---

EDIT: also added `GOLANG_VERSION` to the suggestion to match `golang:trixie` better (try `docker run --rm golang:trixie env`)

---

This is a nice suggestion in terms of fixing the toolchain version. However, I don't see `GOLANG_VERSION` being listed in `docker run --rm golang:trixie env`. This is also a behaviour conflicting with the default behaviour when we have a golang package installed in the system.

Looping @cjdcordeiro for decision.

---

afaiu `GOLANG_VERSION` is not part of standard Go env, so I'd get rid of it.

The `GOTOOLCHAIN` is interesting. Would this rock be able to cope with `local+auto` and successfully download another version of the toolchain if the go.mod says so? 

If this requires more investigation, I'd just go with `local` for now and add `+auto` later, if deemed appropriate

---

Yes, this rock is able to automatically download a newer version of the toolchain if declared in the `go.mod`.

---

Cool. Let's consider that for a follow up PR, where we also add some tests to cover that. For now we can go with `local` only

---

> I don't see GOLANG_VERSION being listed in docker run --rm golang:trixie env

```bash
$ docker run --rm golang:trixie env
PATH=/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=4ee59bba107b
GOLANG_VERSION=1.25.5
GOTOOLCHAIN=local
GOPATH=/go
HOME=/root
```

---

> Cool. Let's consider that for a follow up PR

:+1: 

also let's consider the `GOLANG_VERSION` later to